### DISCUSSION TITLE
Support Blade Icons

### DISCUSSION THREAD

**Discussion by AlexanderFalkenberg** [livewire/flux#232] [Feature Requests]
Add support for the integration of blade icons. Heroicons has a limited set of icons. Filament uses Blade Icons under the hood, making it possible to use different icon sets. There are projects where Heroicons are not enough.

**Comment by mauritskorse** [comment]
I have created a laravel package for adding custom icons to flux: https://github.com/Ympact/flux-icons
It's the first version, so there might still be some bugs ;)
It currently only has the [Tabler](https://tabler.io/icons) icon-set, but other vendors can easily be added, although some tweaking might be necessary to make the icons properly work in Flux.

**Comment by troccoli** [comment]
I wholeheartedly support this feature request. There are currently 83 sets of icons that could be used with Blade. Being able to use them in Flux would be a game changer, similar to being able to customise the colours (but that's another discussion).

I had a look at the `flux:icon` command implementation and it looks like the only difference, to be able to use another set on icons other than Lucide, is to
- provide a callback to download the icon
- provide your own implementation of `generateIconBlade()`

The way how I envisage this to work is:
- install Flux
- install a package for the icon set you want to use (similar to what you to if you want to use Blade icons)
- use the `flux:icon` command to pubish the icons, maybe specifying the ison set, e.g.
  `php artisan flux:icon lucide house, external-link, layout-dashboard`

**Reply by mauritskorse** [reply]
I have now support for 8 icon packs in [ympact/flux-icons](https://github.com/Ympact/flux-icons), and for most icon packs its pretty easy to add. 
Then, using `php artisan flux-icons:build` you can easily build any icon (or all) from a vendor.

**Reply by troccoli** [reply]
I'm the maintaner of the `troccoli/blade-health-icons` package, which is to use the Health Icons in your Blade views (using the Blade UI Kit pakcage).

Unfortunately, as it stands, your package cannot be used for the [HealthIcons](https://healthicons.org/) icon because they categorised the icons, so the source to download an icon from is not always the same.

For example the "Blood A-" icon's path is `icons/svg/filled/blood/blood_a_n.svg`, but "Arm" icon's path is `icons/svg/filled/body/arm.svg`.

This is why I thought a mechanism in Flux to plug-in an icon set would work better. I could create my own package, as I did for the Blade  icons, but I was hoping we could find a more generic solution.

**Reply by mauritskorse** [reply]
That shouldn't be a problem. Flowbite has them categorized as well and works too. It is even possible to adjust the names of icons using a name callback so you would turn `blood_a_n.svg` into `<flux.icon.health.blood-a-negative />` if you would like.

**Reply by troccoli** [reply]
Mmmm, interesting. I'll give it a try and report back.

**Reply by troccoli** [reply]
I don't seem to be able to make your package work, even with the other providers. I will open an issue on your repository, so we can keep the discussion there.

**Reply by mauritskorse** [reply]
Should have fixed the issues you had. In the mean time I added support for the Healthicons as well.

**Comment by calebporzio** [comment]
Going to close this as we've added support for Lucide icons and documentation for adding custom svgs: https://fluxui.dev/components/icon#lucide-icons

Please re-submit a new discussion if people still want blade icons so we can gauge demand.

Thanks!

**Comment by GregoireGaonach** [comment]
As stated in the documentation, publishing a component is not "as scary as it may first seem". In my project, I use Blade Icons extensively, including icon names stored in database columns, which caused some compatibility issues.

By simply publishing `flux/icon/index.blade.php` and adding a basic condition, I was able to automatically use Blade Icons throughout the project.

```php
@if(Str::startsWith($icon, ['lucide-', 'si-']))
    <x-dynamic-component
        data-flux-icon
        {{ $attributes->class($classes) }}
        :component="$icon"
    />
@else
    <flux:delegate-component :component="'icon.' . $icon">{{ $slot }}</flux:delegate-component>
@endif
```
